search/facets_builder.go

```diff
@@ -127,6 +127,9 @@ type TermFacets struct {
 }
 
 func (tf *TermFacets) Terms() []*TermFacet {
+	if tf == nil {
+		return []*TermFacet{}
```

Comment:
Shouldn't  the`tf` instance needs to be initialized correctly and then the default return would send back the right tf.termFacets slice?

Reply:
Oh it's initialized alright - but only when available. This is to handle the case when there aren't any term facets available. As pointed out https://github.com/blevesearch/bleve/issues/1653, previously this didn't need a nil guard - because TermFacets within FacetResult wasn't a pointer but now it is.